Code/SimDivPickers/MaxMinPicker.h

```diff
@@ -133,71 +141,115 @@ RDKit::INT_VECT MaxMinPicker::lazyPick(T &func, unsigned int poolSize,
   if (poolSize < pickSize)
     throw ValueErrorException("pickSize cannot be larger than the poolSize");
 
-  RDKit::INT_LIST pool;
-
   RDKit::INT_VECT picks;
-  picks.reserve(pickSize);
-  unsigned int pick = 0;
 
-  // enter the pool into a list so that we can pick out of it easily
-  for (unsigned int i = 0; i < poolSize; i++) {
-    pool.push_back(i);
-  }
+  unsigned int memsize = (unsigned int)(poolSize * sizeof(MaxMinPickInfo));
+  MaxMinPickInfo *pinfo = (MaxMinPickInfo *)malloc(memsize);
```

Comment:
I suggest using a std::vector here.  malloc has a tendency to lie about allocation anyway.

Reply:
or at least new and delete.